From 4c5b0535f05b8ea5fc1dbd7179d487728edea757 Mon Sep 17 00:00:00 2001
From: Sophie Herold <sophie@hemio.de>
Date: Thu, 20 Apr 2023 09:25:02 +0200
Subject: [PATCH] Fix ICC profile not completely decompressed

Without this fix the icc profile would usually be empty

Origin: upstream, https://github.com/image-rs/image-png/commit/4c5b0535f05b8ea5fc1dbd7179d487728edea757

---
 src/decoder/stream.rs | 1 +
 1 file changed, 1 insertion(+)

--- a/src/decoder/stream.rs
+++ b/src/decoder/stream.rs
@@ -1067,6 +1067,7 @@
                 }
                 buf = &buf[consumed_bytes..];
             }
+            inflater.finish_compressed_chunks(&mut profile)?;
 
             info.icc_profile = Some(Cow::Owned(profile));
             Ok(Decoded::Nothing)
